datastore/threads/threadproxyds.nim

```diff
@@ -51,9 +53,43 @@ type
     queryLock: AsyncLock      # global query lock, this is only really \
                               # needed for the fsds, but it is expensive!
 
+proc addrOf*[T](ctx: ref TaskCtx[T]): ptr TaskCtx[T] =
+  result = cast[ptr TaskCtx[T]](ctx)
+  echo "ADDR_OF: ", result.pointer.repr
+
+proc new*[T](
+    ctx: typedesc[TaskCtx[T]],
+    ds: Datastore,
+): ref TaskCtx[T] =
+  result = (ref TaskCtx[T])()
+  result.ds = unsafeAddr(ds) ##\
+    ## doing this appears to break things. previously it was using `addr(ds)`
+    ## and reverting to those lets the tests get further. 
+    ## 
+    ## however, that seems to mean that `addr(ds)` we're taking the 
+    ## address of the `var ds: Datastore` location, and not the actual
+    ## Datastore:ObjectType. As in `ds: ptr(ref Datastore:ObjectType)`. 
+    ## 
+    ## so doing the `unsafeAddr` would mean we're taking the ptr location
+    ## of the `ds` argument, which is a temporary stack location.
+    ## 
+    ## not sure how to fix this while using GC types
+    ## just taking the address of the var location sorta works, but
+    ## crashes now as well, but later. likely due to datastore now being
```

Comment:
@dryajov related to the question of addr vs baseAddr. I've tried a few variations. But the `ptr Datastore` keeps segfaulting at some point. The `ptr TaskCtx` change I did appears fine and it's contents are what they're set to be.

Reply:
Here's a simplified version: https://github.com/codex-storage/nim-datastore/pull/50 that does the same thing. If you can capture the `var Datastore` address it sorta works, but I think pointers to a bigger issue with the GC type. Doing stress tests last week I saw lots of GC issues with other ref types.